s2i/builder.go

```diff
@@ -120,15 +134,41 @@ func (b *Builder) Build(ctx context.Context, f fn.Function) (err error) {
 	return
 }
 
-// defaultBuilderImage for the given function based on its runtime, or an
-// error if no default is defined for the given runtime.
-func defaultBuilderImage(f fn.Function) (string, error) {
+// builderImage for Function
+// Uses the image defined on the Function by default (for the given runtime)
+// or uses the static defaults if not defined. Returns an  ErrRuntimeRequired
+// if the Function failed to define a Runtime, and ErrRuntimeNotSupported if
+// defined but an image exists neither in the static defaults nor in the
+// Function's Builders map.
+func builderImage(f fn.Function) (string, error) {
 	if f.Runtime == "" {
 		return "", ErrRuntimeRequired
 	}
-	v, ok := DefaultBuilderImages[f.Runtime]
-	if !ok {
-		return "", ErrRuntimeNotSupported
+
+	v, ok := f.BuilderImages["s2i"]
```

Comment:
@lkingland am I reading this correctly that if there is an s2i builder image defined, it will be used as the default?